gapis/api/vulkan/profile_static_analysis.go

```diff
@@ -77,44 +373,144 @@ func (API) ProfileStaticAnalysis(ctx context.Context, p *path.Capture) (interfac
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
 		case VkCmdDrawIndexedArgsʳ:
-			sampler := data.newSampler(idx)
-			data.addSample(sampler, vertexCount, float64(args.InstanceCount()*args.IndexCount()))
-		case VkCmdDrawIndexedIndirectArgsʳ:
+			isIndexedDraw = true
+			vertexCount, primitiveCount, err = getDrawIndexedCounts(ctx, s, args.IndexCount(), args.InstanceCount(), args.FirstIndex())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
 		case VkCmdDrawIndirectArgsʳ:
-		case VkCmdDrawIndirectCountKHRArgsʳ:
-		case VkCmdDrawIndexedIndirectCountKHRArgsʳ:
+			vertexCount, primitiveCount, err = getDrawIndirectCounts(ctx, s, args.Buffer(), args.Offset(), args.DrawCount(), args.Stride())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
+		case VkCmdDrawIndexedIndirectArgsʳ:
+			isIndexedDraw = true
+			vertexCount, primitiveCount, err = getDrawIndexedIndirectCounts(ctx, s, args.Buffer(), args.Offset(), args.DrawCount(), args.Stride())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
+		case VkCmdDrawIndirectCountKHRArgsʳ: // TODO VULKAN_1_2: Add core 1.2 VkCmdDrawIndirectCountArgsʳ
+			drawCount, err := getIndirectCount(ctx, s, args.CountBuffer(), args.CountBufferOffset(), args.MaxDrawCount())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
+			vertexCount, primitiveCount, err = getDrawIndirectCounts(ctx, s, args.Buffer(), args.Offset(), drawCount, args.Stride())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
 		case VkCmdDrawIndirectCountAMDArgsʳ:
+			drawCount, err := getIndirectCount(ctx, s, args.CountBuffer(), args.CountBufferOffset(), args.MaxDrawCount())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
+			vertexCount, primitiveCount, err = getDrawIndirectCounts(ctx, s, args.Buffer(), args.Offset(), drawCount, args.Stride())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
+		case VkCmdDrawIndexedIndirectCountKHRArgsʳ: // TODO VULKAN_1_2: Add core 1.2 VkCmdDrawIndexedIndirectCountArgsʳ
+			isIndexedDraw = true
+			drawCount, err := getIndirectCount(ctx, s, args.CountBuffer(), args.CountBufferOffset(), args.MaxDrawCount())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
+			vertexCount, primitiveCount, err = getDrawIndexedIndirectCounts(ctx, s, args.Buffer(), args.Offset(), drawCount, args.Stride())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
 		case VkCmdDrawIndexedIndirectCountAMDArgsʳ:
+			isIndexedDraw = true
+			drawCount, err := getIndirectCount(ctx, s, args.CountBuffer(), args.CountBufferOffset(), args.MaxDrawCount())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
+			vertexCount, primitiveCount, err = getDrawIndexedIndirectCounts(ctx, s, args.Buffer(), args.Offset(), drawCount, args.Stride())
+			if err != nil {
+				log.W(ctx, err.Error())
+				return
+			}
+		default:
+			// Not implemented
```

Comment:
Should we put an error here or something?

Reply:
Done.